### PR TITLE
os/board/rtl8730e: Fix Peripheral related Build Warning and SVACE

### PR DESCRIPTION
**Pull request by ZhenBei-Sin**
- Deleted the unused API and sources file that caused Build Warning and SVACE 
Build Warning
- Added return for the function does return
- Removed the unused variable
- Performed the correct casting conversion
- Solved the shadowed declaration without use the same variable name for the global and local
- Added the define/undefine condition checking for the variable 
SVACE
- DREF_OF_NULL.ASSIGN : Performed NULL checking
- SIMILAR_BRANCHES : Removed the idetical branches and keep one only
- NO_CAST.INTEGER_DIVISION & INTEGER_OVERFLOW : Performed the correct casting conversion
- Added the comment for unable solve UNREACHEABLE_CODE

### CODE REVIEW

**Review by ewoodev — APPROVED** [review]
Looks good to me

**Review by sunghan-chang — DISMISSED** [review]

File: os/board/rtl8730e/src/component/mbed/targets/hal/rtl8730e/us_ticker.c [hunk 1]
```diff
@@ -63,7 +63,7 @@ uint32_t us_ticker_read(void)
 	uint64_t us_tick;
 
 	tick_cnt = SYSTIMER_TickGet(); //up counter
-	us_tick = tick_cnt * (1000000 / 32768);
+	us_tick = (uint64_t) (tick_cnt * (1000000 / 32768));
```
Comment (sunghan-chang) [inline os/board/rtl8730e/src/component/mbed/targets/hal/rtl8730e/us_ticker.c:66]: Is this really working? Not `(uint64_t)tick_cnt × (1000000 / 32768)`?

Comment (ZhenBei-Sin) [reply os/board/rtl8730e/src/component/mbed/targets/hal/rtl8730e/us_ticker.c]: Hi @sunghan-chang ,

I apologize for the mistake. You're correct.
After verified :
The suitable way is to cast the tick_cnt to uint64_t.

image

Done git push --force the changes.
Thank You.

File: os/board/rtl8730e/src/component/mbed/targets/hal/rtl8730e/pwmout_api.c [hunk 2]
```diff
@@ -132,7 +132,7 @@ void pwmout_init(pwmout_t *obj, PinName pin)
 
 	obj->pwm_idx = pwm_chan;
 	obj->period = 0x10000 * (prescaler + 1) / 40;
-	obj->pulse = 0x1000 * (prescaler + 1) / 40;
+	obj->pulse = (float) (0x1000 * (prescaler + 1) / 40);
```
Comment (sunghan-chang) [inline os/board/rtl8730e/src/component/mbed/targets/hal/rtl8730e/pwmout_api.c:135]: @ZhenBei-Sin Could you check this typecasting as well? It's the same as previous comment.

Comment (ZhenBei-Sin) [reply os/board/rtl8730e/src/component/mbed/targets/hal/rtl8730e/pwmout_api.c]: Changed and Verified:

image

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**